model/task_config.go

```diff
@@ -22,7 +23,7 @@ type TaskConfig struct {
 	WorkDir      string
 }
 
-func NewTaskConfig(d *distro.Distro, v *version.Version, p *Project, t *task.Task, r *ProjectRef) (*TaskConfig, error) {
+func NewTaskConfig(d *distro.Distro, v *version.Version, p *Project, t *task.Task, r *ProjectRef, patchDoc *patch.Patch) (*TaskConfig, error) {
```

Comment:
can't you get the patch document from the task document? if so, why do we need to pass it in?

Reply:
I'm not aware of any way to get the patch document from the task. Is there a specific method you're thinking of?

Reply:
somehow, the API is doing it in GetTaskPatch, so I presume its possible, presumably (though I'm not sure,) the version on the task is the patchid